### ISSUE TITLE
Support status code for file response

### ISSUE THREAD

**Issue by FelixZY** [channelcat/sanic#1268]
## TL;DR
It is not possible to set response status for `response.file` and `response.file_stream` as currently suggested in documentation

## Issue
Current documentation e.g. at http://sanic.readthedocs.io/en/latest/sanic/response.html#modify-headers-or-status states that you simply need to add the `status` argument to set a custom status code for a response. This is not the case however for `response.file` and `response.file_stream` which always return status 200 by default (`sanic/response.py:263`, `sanic/response.py:318`).

This became an issue today where I wanted to show a login page for an unauthorized user but still return status 401.

Expected code:
```python
@app.exception(Unauthorized)
async def handle_401(request, exception):
    return response.file(web_dir / "login.html", status=401)
```

Actual code:
```python
@app.exception(Unauthorized)
async def handle_401(request, exception):
    response = await response.file(web_dir / "login.html")
    response.status = 401
    return response
```
## Solution
Looking at the source, it seems simple enough to add a `status` argument and use its value for `response.file` and `response.file_stream` like so:

```python
# response.py:236
async def file(
        location, status=200, mime_type=None, headers=None, filename=None, _range=None):
    """Return a response object with file data.
    :param location: Location of file on system.
    :param status: Response code.
    :param mime_type: Specific mime_type.
    :param headers: Custom Headers.
    :param filename: Override filename.
    :param _range:

    # [...]

    # response.py:263
    return HTTPResponse(status=status,
                        headers=headers,
                        content_type=mime_type,
                        body_bytes=out_stream)
```

However, if there are other considerations involved it would be good to explicitly state `status` is not available for `response.file` and `response.file_stream` in documentation

**Comment by MyNameIsCosmo** [comment]
You should be able to use `StreamingHTTPResponse` as so:
```python
@app.exception(Unauthorized)
async def handle_401(request, exception):
    return response.StreamingHTTPResponse(web_dir / "login.html", status=401)
```

As for defining status in `response.file` and `response.file_stream`, check out PR #1269
